python/ray/dag/compiled_dag_node.py

```diff
@@ -831,6 +829,30 @@ def _preprocess(self) -> None:
                         )
                     direct_input = True
 
+                elif isinstance(upstream_node.dag_node, ClassMethodNode):
+                    from ray.dag.constants import RAY_ADAG_ENABLE_DETECT_DEADLOCK
+
+                    if (
+                        # Ray aDAG deadlock detection has the same check, but
+                        # it may be turned off because of false positives.
+                        # In that case, we need this check to be active.
+                        # TODO: When we clean up Ray aDAG deadlock detection
+                        # this check should be done at one place only.
+                        not RAY_ADAG_ENABLE_DETECT_DEADLOCK
+                        and downstream_actor_handle is not None
+                        and downstream_actor_handle
+                        == upstream_node.dag_node._get_actor_handle()
+                        and upstream_node.dag_node.type_hint.requires_nccl()
+                    ):
+                        raise ValueError(
+                            "Compiled DAG does not support NCCL communication between "
+                            "methods on the same actor. NCCL type hint is specified "
+                            "for the channel from method "
+                            f"{upstream_node.dag_node.get_method_name()} to the method "
```

Comment:
Can you also include actor name and their id? 

Actor.fwd -> Actor.bwd (ID: XXX).

Reply:
Done. Added actor handle which contains both info.